nautobot_floor_plan/static/nautobot_floor_plan/js/floorplan.js

```diff
@@ -290,134 +304,426 @@ document.addEventListener('DOMContentLoaded', function() {
+    
+            // Compute width and height safely
+            let computedW = isNaN(endX - startX) ? 100 : Math.max(10, endX - startX);
+            let computedH = isNaN(endY - startY) ? 100 : Math.max(10, endY - startY);
+    
+            let newW = Math.min(safeOrigW, computedW);
+            let newH = Math.min(safeOrigH, computedH);
+    
+            console.log(`Final Computed ViewBox: ${startX} ${startY} ${newW} ${newH}`)
+
+            // Disable zoom and pan during highlight operation
+            disableZoomAndPan();        
+            // Apply zoom with animation
+            if (!isNaN(startX) && !isNaN(startY) && !isNaN(newW) && !isNaN(newH)) {
+                gsap.to(svg, { 
+                    duration: 1, 
+                    attr: { viewBox: `${startX} ${startY} ${newW} ${newH}` }, 
+                    ease: "power2.inOut"
+                });
+
+                // Apply highlight effects
+                const effects = createHighlightEffects(element, svg);
+    
+                // Restore original view after 5s
+                setTimeout(() => {
+                    console.log("Resetting viewBox using resetZoom() function.");
+                    resetZoom(); // Call the existing reset function
+                    // Re-enable zoom and pan after highlight is complete
+                    enableZoomAndPan();
+                }, 5000);
+    
+                // Clean up effects after 20s
+                setTimeout(() => {
+                    console.log("Cleanup triggered after 20 seconds");
+                    completeCleanup(element, effects.elements, effects.animations);                
+                }, 20000);
+            } else {
+                console.error("Invalid viewBox values. Skipping update.");
+            }
+            // Scroll to the SVG container after zooming
+            const anchorElement = document.getElementById("zoom-anchor");
+            if (anchorElement) {
+                // Ensure scroll happens after zoom animation
+                gsap.delayedCall(1.1, () => {  // Delay to ensure zoom is complete
+                    anchorElement.scrollIntoView({ behavior: "smooth", block: "start" });
+                });
+            }
+        }
+    }
+    // Disable zoom and pan interactions
+    function disableZoomAndPan() {
+        // Temporarily remove mouse listeners for zoom and pan
+        document.querySelector("#floor-plan-svg").style.pointerEvents = "none"; // Disable pointer events
+    }
+
+    // Enable zoom and pan interactions
+    function enableZoomAndPan() {
+        // Restore mouse listeners for zoom and pan
+        document.querySelector("#floor-plan-svg").style.pointerEvents = "auto"; // Re-enable pointer events
+    }
+    // Function for complete cleanup of all highlighting effects
+    function completeCleanup(element, effectElements, animations) {
+        console.log("Performing complete cleanup of highlighting");
+    
```

Comment:
Nit - can you squash the trailing whitespace on this file? I dig [this extension](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces) - it adds a visual studio action option and highlights it for you in the file visually.